storey/targets.py

```diff
@@ -1124,19 +1123,24 @@ def __init__(
             kwargs["infer_columns_from_data"] = infer_columns_from_data
         Flow.__init__(self, **kwargs)
         _Writer.__init__(self, columns, infer_columns_from_data, retain_dict=True)
-        self._table = table
         if isinstance(table, str):
-            if not self.context:
-                raise TypeError("Table can not be string if no context was provided to the step")
-            self._table = self.context.get_table(table)
-        self._closeables = [self._table]
+            self._table_name = table
+            self._table = None
+        else:
+            self._table_name = None
+            self._table = table
 
         self._field_extractor = lambda event_body, field_name: event_body.get(field_name)
         self._write_missing_fields = False
 
     def _init(self):
         Flow._init(self)
         _Writer._init(self)
+        if self._table_name:
+            if not self.context:
+                raise TypeError("Table can not be string if no context was provided to the step")
+            self._table = self.context.get_table(self._table_name)
+            self._closeables = [self._table]
```

Comment:
Will add a test to cover this use case (reuse of `NoSqlTarget`.

Reply:
Reverted that change for now. It's more involved and risky than the rest of this PR, and doesn't relate to the ticket in question.